src/Backend/AMQPBackendDispatcher.php

```diff
@@ -133,7 +133,22 @@ public function getChannel()
             ]);
 
             if ($factory instanceof DelayStrategyAware) {
-                $factory->setDelayStrategy(new RabbitMqDlxDelayStrategy());
+                $delayStrategy = $this->settings['delay_strategy_class'];
+                if (
+                    !empty($delayStrategy) &&
+                    (
+                        !class_exists($delayStrategy) ||
+                        !(new \ReflectionClass($delayStrategy))->implementsInterface(DelayStrategy::class)
```

Comment:
:thinking: why not just
```suggestion
                        !($delayStrategy instanceof DelayStrategy::class)
```
?

Reply:
I can change it. I just kept the same check that is done for the `factory_class`.

Reply:
Oh ok… that's weird. If you can have a look at the history/find the PR and ultimately the reason for this, great. If not, please just change it.

Reply:
The entire `getContext()` function was created as is when implementing the new interop library. I don't think this was done for BC purposes. 

You can look at the entire change set at #276 

I am just changing it.

Reply:
@greg0ire I am stupid lol. The parameter is a class name (string) so it cannot be instanceof anything. This is why you use the ReflectionClass to assert whether or not it implements it. 

Another way would be to factor the if a little (basic algebraic principles) and make a new object in the meantime to do an instanceof. But that really makes the condition to be mathematical and becomes less human readable. 

I am going to keep it as it is.

Reply:
Oh right. But there is still no need for Reflection, you could simply use https://php.net/is_a , with the third argument set to `true`